pkg/antctl/raw/migrate/convertnetworkpolicy/command.go

```diff
@@ -0,0 +1,488 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package networkpolicy
+
+package convertnetworkpolicy
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	calicoclient "github.com/projectcalico/api/pkg/client/clientset_generated/clientset"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/antctl/raw"
+	"antrea.io/antrea/pkg/antctl/raw/migrate"
+	"antrea.io/antrea/pkg/apis/crd/v1beta1"
+	antrea "antrea.io/antrea/pkg/client/clientset/versioned"
+)
+
+const (
+	convertFromCalicoAnnotation = "convert-from-calico"
+
+	defaultPriority = 1
+)
+
+type convertOptions struct {
+	from string
+}
+
+type converter struct {
+	antreaClient *antrea.Clientset
+	calicoClient *calicoclient.Clientset
+}
+
+var (
+	Command *cobra.Command
+	options *convertOptions
+)
+
+func (o *convertOptions) validateAndComplete() error {
+	if o.from != "calico" {
+		return fmt.Errorf("unsupported CNI: %s", o.from)
+	}
+	return nil
+}
+
+func newConverter(kubeconfig *rest.Config) (*converter, error) {
+	_, antreaClient, err := raw.SetupClients(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	calicoClient, err := calicoclient.NewForConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	return &converter{
+		antreaClient: antreaClient,
+		calicoClient: calicoClient,
+	}, nil
+}
+
+var migrateCommandExample = strings.Trim(`
+  Convert Calico NetworkPolicies to Antrea NetworkPolicies:
+  $ antctl migrate convert-networkpolicy --from calico
+`, "\n")
+
+func init() {
+	Command = &cobra.Command{
+		Use:     "convert-networkpolicy",
+		Short:   "Convert NetworkPolicy from other CNI to Antrea",
+		Long:    "Convert NetworkPolicy from other CNI to Antrea",
+		Example: migrateCommandExample,
+		RunE:    runE,
+		Args:    cobra.NoArgs,
+	}
+
+	o := &convertOptions{}
+	options = o
+	Command.Flags().StringVar(&o.from, "from", "", "The CNI to migrate from. Supported values: calico.")
+}
+
+func runE(cmd *cobra.Command, _ []string) error {
+	if err := options.validateAndComplete(); err != nil {
+		return err
+	}
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	c, err := newConverter(kubeconfig)
+	if err != nil {
+		return err
+	}
+	if err := c.convertNetworkPolicy(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *converter) convertNetworkPolicy() error {
+	if options.from == "calico" {
+		return c.convertCalicoNetworkPolicy()
+	}
+	return nil
+}
```

Comment:
I feel it's unnecessary to have this function, you can move  lines 119-122 to line 111 directly.